src/shared/parameter_v2/c/config_test.cpp

```diff
@@ -0,0 +1,106 @@
+#include "shared/parameter_v2/c/parameter.h"
+}
+
+#include "software/util/variant_visitor/variant_visitor.h"
+#include "yaml-cpp/yaml.h"
+
+class YamlLoadFixture : public ::testing::Test
+{
+   protected:
+    typedef struct ConfigMetadata
+    {
+        std::vector<std::string> included_configs;
+        std::vector<YAML::Node> parameters;
+    } ConfigMetadata_t;
+
+    /*
+     * Sets up the test by loading the yaml files in the config folder.
+     *
+     * NOTE: These tests are highly coupled with the generate_dynamic_parameters.py script
+     * and the location of the config folder.
+     */
+    void SetUp() override
+    {
+        // this is loaded from bazel data
+        boost::filesystem::path path("./shared/parameter_v2/config/test");
+
+        for (auto& entry : boost::filesystem::directory_iterator(path))
+        {
+            std::vector<YAML::Node> config = YAML::LoadAllFromFile(entry.path().string());
+            ASSERT_NE(config.size(), 0);
+            ConfigMetadata_t metadata;
+
+            if (config.size() == 1)
+            {
+                try
+                {
+                    // Each yaml file is separated by a --- which indicates the end
+                    // of a yaml document. A new document can start below ---.
+                    // Each config can have 1 or 2 documents, which may contain just
+                    // includes, just parameters (1 document) or includes and parameters
+                    // (2 documents).
+                    //
+                    // We try to load the first document as a parameter definitions, and
+                    // if the conversion doesn't match, we load it as an include list.
+                    metadata.parameters = config[0].as<std::vector<YAML::Node>>();
+                }
+                catch (const YAML::BadConversion& e)
+                {
+                    metadata.included_configs =
+                        config[0]["include"].as<std::vector<std::string>>();
+                }
+            }
+
+            if (config.size() == 2)
+            {
+                // load both the documents because they exist
+                // we don't need to worry too much about format/checking validity
+                // as the schema check should have done that already.
+                metadata.included_configs =
+                    config[0]["include"].as<std::vector<std::string>>();
+                metadata.parameters = config[1].as<std::vector<YAML::Node>>();
+            }
+
+            config_file_to_metadata_map[entry.path().string()] = metadata;
+        }
+    }
+
+    // store the config metadata which contains the included configs and the parameters
+    // in a mapping to the config file it was loaded from
+    std::map<std::string, ConfigMetadata_t> config_file_to_metadata_map;
+};
+
+TEST_F(YamlLoadFixture, MemoryInitializationSanityCheck)
+{
+    const ThunderbotsConfig_t* config = app_dynamic_parameters_create();
```

Comment:
maybe we need memory leak checks on this?

Reply:
added to ci checks